### PR TITLE
DHFPROD-5764: Data Service/REST API for previewing match step against…

### PR DESCRIPTION
**Pull request by fsnow**
… documents

NO NEED FOR REVIEW, just want to run unit tests.

#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests
  

- ##### Reviewer:

- [ ] Reviewed Tests

### CODE REVIEW

**Review by Sanjeevani19 — COMMENTED** [review]
@fsnow the pipeline wouldn't run tests if it is in draft state so changed it to review stage.

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/mastering/previewMatchingActivity.api [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+  {
+    "datatype": "jsonDocument",
+    "$javaClass": "com.fasterxml.jackson.databind.JsonNode",
+    "description": "Returns the documentType with an 'id' matching the input",
+    "schema":
+    {
+      "$ref": "#/components/schemas/MatchActivityPreview"
+    }
+  },
+  "components":
```
Comment (rjrudin) [inline marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/mastering/previewMatchingActivity.api:27]: Does ML do anything with this? I wasn't aware that this is a thing

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/mastering/previewMatchingActivity.sjs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+var stepName;
+var uris;
+var sampleSize;
+
+xdmp.securityAssert("http://marklogic.com/data-hub/privileges/read-match-merge", "execute");
+
+
+const step = require('/data-hub/5/artifacts/core.sjs').getArtifact("matching", stepName);
+
+let result = previewMatchingActivityLib.previewMatchingActivity(step, uris, sampleSize);
```
Comment (rjrudin) [inline marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/mastering/previewMatchingActivity.sjs:29]: I know this is a draft, but one thought here - I wouldn't make a library until one is needed, and once one is needed, let's put it under /data-hub/5/mastering .

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/mastering/previewMatchingActivity.api (hunk 1 above)
Comment (fsnow) [reply marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/mastering/previewMatchingActivity.api:27]: Good question. I pasted this into the api file from the spec doc:
https://wiki.marklogic.com/display/ENGINEERING/Hub+Central+Mastering+Design+Specification

**Pull request closed without merging**